### ISSUE TITLE
Blank Page on Debian 8

### ISSUE THREAD

**Issue by robingroppe** [Typesetter/Typesetter#140]
After filling the installation form a blank page appears and installation fails. No php errors.
When migrating an installed sited from ubuntu to debian login fails. No php errors either.

Debian 8, Apache 2.4 with php-fpm, php 5.6 using imscp.

:(

**Comment by robingroppe** [comment]
Someone there?

**Comment by 2braincells2go** [comment]
I would recommend trying at http://www.typesettercms.com/Forum

**Comment by oyejorge** [comment]
A blank page to me says there's either a blank index.html file (which seems
unlikely) or a phone error. I know you said there weren't any php errors
but did you check the error log?

**Comment by thomasfrobieter** [comment]
I agree, take look at the error log of your apache server. You may have disabled the error output?

**Comment by robingroppe** [comment]
Its not disabled as other errors are beeing logged. There are no other
index files. This problem occures when using php-fpm only. On a test setup
on a rpi it runs fine.
This is a huge problem because on my main server its essential to use fpm.
I could provide a test enviroment.
Am 09.06.2016 07:30 schrieb "Thomas" <notifications@github.com>:

**Comment by thomasfrobieter** [comment]
Okay, so it´s not a simple "noob failure" ;) 
Then i recommend pasting your question into the forum, as theres a lot more activity: https://www.typesettercms.com/Forum?show=f10

**Comment by seidler2547** [comment]
It's because of [Debian bug 795572](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=795572). For some reason, the Debian developers decided they needed to [implement getallheaders](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=742497) but do so in an incomplete way.

In any case, I also don't understand the need to check for the `Content-Length` headers twice in `include/tool.php` (around line 283, `public static function CheckRequest()`). I would rather think that the logic should be:
```
if( !isset($_SERVER['CONTENT_LENGTH']) ){
  if( function_exists('getallheaders') ){
    if( !isset($headers['Content-Length']) ){
      die();
    } else {
      return;
    }
  }
  die();
}
```
So only use `getallheaders` if `$_SERVER['CONTENT_LENGTH']` is not present.

And also, some error logging at that place would be really **really** nice because it took me almost two days finding this!

**Comment by seidler2547** [comment]
Is there anything we can do to fix or work around this?

**Comment by mckaygerhard** [comment]
umm i do not see that but this happened me sinde i moved to php 5.6+ in older debian 6 and php 5.4 does not happened ! 

do you tried @seidler2547  by changed the "die()=" ?

**Comment by juek** [comment]
allegedly fixed as of PHP 5.6.26
